### PR TITLE
Setup of category seed first

### PR DESCRIPTION
**Pull request by carolynyoo**
Zeke's suggestion to create categories first then find/use those _ids.

### CODE REVIEW

**Comment by carolynyoo** [comment]
Issue #39

**Comment by carolynyoo** [comment]
This seed works for Film, Category, User so should be good to go/useable for most stuff. 

Review collection is still giving an error like so 
`{ [CastError: Cast to ObjectId failed for value "Obama" at path "user"]
  message: 'Cast to ObjectId failed for value "Obama" at path "user"',
  name: 'CastError',
  type: 'ObjectId',
  value: 'Obama',
  path: 'user' }` --> ARGH

Feel free to merge after review and please override previous file with this wholly when resolving merge conflict. I accounted for review model/Randy's fixes.

File: seed.js [hunk 1]
```diff
@@ -112,92 +130,87 @@ var seedFilms = function () {
...
+.then(function (reviews) {
+    console.log(reviews);
+})
+.then(function () {
+    console.log(chalk.blue('All seeds successful!'));
+    process.kill(0);
+})
+.catch(function (err) {
+    console.error(err);
+    process.kill(1);
```
Comment (itsaliu90) [inline seed.js]: Can you explain why we need a 'process.kill()' line?

File: seed.js [hunk 2]
```diff
@@ -24,47 +24,65 @@ var connectToDb = require('./server/db');
...
-        {
-            email: 'testing@fsa.com',
-            name: 'Testing',
-            password: 'password'
-        },
-        {
-            email: 'obama@gmail.com',
-            name: 'Obama',
-            password: 'potus'
+var findIdFromDb = function (elem, db, keyName) {
```
Comment (itsaliu90) [inline seed.js:42]: This is a cool function.

File: seed.js [hunk 3]
```diff
@@ -112,92 +130,87 @@ var seedFilms = function () {
...
-    });
-});
+connectToDb.then(function () {
+    return wipeDb()
+})
+.then(function() {
+    console.log(chalk.blue('Wiped database'));
+})
+.then(function() {
+    q.spread([createFilms, createUsers], function (films, users) {
```
Comment (itsaliu90) [inline seed.js:194]: Way to rock q, you'll have to give us a mini-lecture on that :).

**Comment by itsaliu90** [comment]
:hamburger:

File: seed.js (hunk 1 above)
Comment (carolynyoo) [inline seed.js]: a nice to have so that ctrl-c isn't needed. 0 or 1 indicates success/error

File: seed.js (hunk 2 above)
Comment (carolynyoo) [inline seed.js:42]: :heart_eyes:

**Comment by carolynyoo** [comment]
OMG IT ALL WORKS NOW I CAN SLEEP IN PEACE :zzz: :sleeping: 

screen shot 2015-04-25 at 12 26 33 am

Note to self: chain promises all the way instead of mixing nesting & chaining.

File: seed.js [hunk 4]
```diff
@@ -112,92 +129,85 @@ var seedFilms = function () {
...
+        return q.invoke(Category, 'create', seedCategories);
+    })
+    .then(function () {
+        return Category.find().exec();
+    })
+    .then(function (categories) {
+        console.log(chalk.green('Category seed successful!'));
+        seedFilms.forEach(function (film, filmIndex, filmArr) {
+            film['categories'].forEach(function (elem, index, arr) {
+                var filmID = findIdFromDb(elem, categories, 'name');
```
Comment (gskerry) [inline seed.js]: should this be 'categoryId'? That's what's getting returned here, right? Trying to follow chain...

Comment (carolynyoo) [inline seed.js]: yea oops. will fix

**Pull request closed without merging**